tools/kagami/CLI.md

```diff
@@ -0,0 +1,315 @@
+
+  Default value: `0`
+* `--assets-per-domain <ASSETS_PER_DOMAIN>` — Number of assets per domains in synthetic genesis. Total number of assets would be `domains * assets_per_domain`
+
+  Default value: `0`
+
+
+
+## `kagami config`
+
+Generate the default client/peer configuration
+
+**Usage:** `kagami config <COMMAND>`
+
+###### **Subcommands:**
+
+* `client` — 
+* `peer` — 
+
+
+
+## `kagami config client`
+
+**Usage:** `kagami config client`
+
+
+
+## `kagami config peer`
+
+**Usage:** `kagami config peer`
+
+
+
+## `kagami docs`
+
+Generate a Markdown reference of configuration parameters
+
+**Usage:** `kagami docs`
+
+
+
+## `kagami tokens`
+
+Generate a list of predefined permission tokens and their parameters
+
+**Usage:** `kagami tokens`
+
+
+
+## `kagami validator`
+
+Generate the default validator
+
+**Usage:** `kagami validator`
+
+
+
+## `kagami swarm`
+
+Generate a docker-compose configuration for a variable number of peers
+using a Dockerhub image, GitHub repo, or a local Iroha repo.
+
+This command builds the docker-compose configuration in a specified directory. If the source
+is a GitHub repo, it will be cloned into the directory. Also, the default configuration is
+built and put into `<target>/config` directory, unless `--no-default-configuration` flag is
+provided. The default configuration is equivalent of running:
+
+```bash
+kagami config peer
+kagami validator
+kagami genesis default --compiled-validator-path ./validator.wasm
+```
+
+Default configuration building will fail if Kagami is run outside of Iroha repo ([tracking
+issue](https://github.com/hyperledger/iroha/issues/3473)). If you are going to run it outside
+of the repo, make sure to pass `--no-default-configuration` flag.
+
+Be careful with specifying a Dockerhub image as a source: Kagami Swarm only guarantees that
+the docker-compose configuration it generates is compatible with the same Git revision it
+is built from itself. Therefore, if specified image is not compatible with the version of Swarm
+you are running, the generated configuration might not work.
+
+**Usage:** `kagami swarm [OPTIONS] --peers <PEERS> --outdir <OUTDIR> <--image <IMAGE>|--build <PATH>|--build-from-github>`
+
+###### **Options:**
+
+* `--image <IMAGE>` — Use specified docker image
+* `--build <PATH>` — Use local path location of the Iroha source code to build images from
+* `--build-from-github` — Clone `hyperledger/iroha` repo from the revision Kagami is built itself, and use the cloned source code to build images from
+* `-p`, `--peers <PEERS>` — How many peers to generate within the docker-compose
+* `--outdir <OUTDIR>` — Target directory where to place generated files
+* `--outdir-force` — Re-create the target directory if it already exists
+* `--no-default-configuration` — Do not create default configuration in the `<outdir>/config` directory
+* `-s`, `--seed <SEED>` — Might be useful for deterministic key generation
```

Comment:
Here is the example of that `clap-markdown` doesn't render additional documentation for arguments. Even if I put `#[clap(verbatim_doc_comment)]`.

For example, `--no-default-configuration`'s full documentation is as follows:

> Do not create default configuration in the `<outdir>/config` directory.
> 
> Default `config.json`, `genesis.json` and `validator.wasm` are generated and put into
> the `<outdir>/config` directory. That directory is specified in the Docker Compose
> `volumes` field.
> 
> If you don't need the defaults, you could set this flag. The `config` directory will be
> created anyway, but you should put the necessary configuration there by yourself.
>

Reply:
Ah, now I see it, thanks. So it doesn't recurse into the submodules args fully. That should be trivially fixable with a home-brew implementation (and sounds really close to what `Documented` does already anyways).